code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,325 @@
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/spider_specialattack(mob/living/carbon/human/L, var/poisonable)
+	L.attack_animal(src)
+
+
+
+
+
+
+// --------------------------------------------------------------------------------
+// --------------------- TERROR SPIDERS: PROC OVERRIDES ---------------------------
+// --------------------------------------------------------------------------------
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/examine(mob/user)
+	..()
+	var/list/msgs = list()
+	if(stat == DEAD)
+		msgs += "<span class='deadsay'>It appears to be dead.</span>\n"
+	else
+		if(key)
+			msgs += "<span class='warning'>Its eyes regard you with a curious intelligence.</span>"
+		if(health > (maxHealth*0.95))
+			msgs += "<span class='notice'>It is in excellent health.</span>"
+		else if(health > (maxHealth*0.75))
+			msgs += "<span class='notice'>It has a few injuries.</span>"
+		else if(health > (maxHealth*0.55))
+			msgs += "<span class='warning'>It has many injuries.</span>"
+		else if(health > (maxHealth*0.25))
+			msgs += "<span class='warning'>It is barely clinging on to life!</span>"
+		else if(health < maxHealth && regen_points > regen_points_per_kill)
+			msgs += "<span class='notice'>It appears to be regenerating quickly</span>"
+		if(killcount == 1)
+			msgs += "<span class='warning'>It is soaked in the blood of its prey.</span>"
+		if(killcount > 1)
+			msgs += "<span class='warning'>It is soaked with the blood of [killcount] prey it has killed.</span>"
+	to_chat(usr,msgs.Join("<BR>"))
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/New()
+	..()
+	ts_spiderlist += src
+	add_language("TerrorSpider")
+	add_language("Galactic Common")
+	default_language = all_languages["TerrorSpider"]
+
+	web_action = new()
+	web_action.Grant(src)
+	wrap_action = new()
+	wrap_action.Grant(src)
+
+	name += " ([rand(1, 1000)])"
+	msg_terrorspiders("[src] has grown in [get_area(src)].")
+	if(is_away_level(z))
+		spider_awaymission = 1
+		ts_count_alive_awaymission++
+	else
+		ts_count_alive_station++
+	// after 30 seconds, assuming nobody took control of it yet, offer it to ghosts.
+	spawn(150) // deciseconds!
+		CheckFaction()
+	spawn(300) // deciseconds!
+		if(spider_awaymission)
+			return
+		if(stat == DEAD)
+			return
+		if(ckey)
+			var/image/alert_overlay = image('icons/mob/terrorspider.dmi', icon_state)
+			notify_ghosts("[src] has appeared in [get_area(src)]. (already player-controlled)", source = src, alert_overlay = alert_overlay)
+		else if(ai_playercontrol_allowingeneral && ai_playercontrol_allowtype)
+			var/image/alert_overlay = image('icons/mob/terrorspider.dmi', icon_state)
+			notify_ghosts("[src] has appeared in [get_area(src)].", enter_link = "<a href=?src=\ref[src];activate=1>(Click to control)</a>", source = src, alert_overlay = alert_overlay, attack_not_jump = 1)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/Destroy()
+	ts_spiderlist -= src
+	return ..()
```

Comment:
You're right. Destroy() does not call Death(). I could duplicate the code from Death(), but I think just calling Death(0) is better.